python/cudf/cudf/core/df_protocol.py

```diff
@@ -0,0 +1,825 @@
+        ``(kind, bit-width, format string, endianness)``
+
+        Kind :
+
+            - INT = 0
+            - UINT = 1
+            - FLOAT = 2
+            - BOOL = 20
+            - STRING = 21   # UTF-8
+            - DATETIME = 22
+            - CATEGORICAL = 23
+
+        Bit-width : the number of bits as an integer
+        Format string : data type description format string in Apache Arrow C
+                        Data Interface format.
+        Endianness : current only native endianness (``=``) is supported
+
+        Notes:
+
+            - Kind specifiers are aligned with DLPack where possible
+             (hence the jump to 20, leave enough room for future extension)
+            - Masks must be specified as boolean with either bit width 1
+             (for bit masks) or 8 (for byte masks).
+            - Dtype width in bits was preferred over bytes
+            - Endianness isn't too useful, but included now in case
+              in the future we need to support non-native endianness
+            - Went with Apache Arrow format strings over NumPy format strings
+              because they're more complete from a dataframe perspective
+            - Format strings are mostly useful for datetime specification,
+              and for categoricals.
+            - For categoricals, the format string describes the type of the
+              categorical in the data buffer. In case of a separate encoding
+              of the categorical (e.g. an integer to string mapping),
+              this can be derived from ``self.describe_categorical``.
+            - Data types not included: complex, Arrow-style null,
+              binary, decimal, and nested (list, struct, map, union) dtypes.
+        """
+        dtype = self._col.dtype
+
+        # For now, assume that, if the column dtype is 'O' (i.e., `object`),
+        # then we have an array of strings
+        if not isinstance(dtype, cudf.CategoricalDtype) and dtype.kind == "O":
+            return (_DtypeKind.STRING, 8, "u", "=")
+
+        return self._dtype_from_cudfdtype(dtype)
+
+    def _dtype_from_cudfdtype(self, dtype) -> ProtoDtype:
+        """
+        See `self.dtype` for details.
+        """
+        # Note: 'c' (complex) not handled yet (not in array spec v1).
+        #       'b', 'B' (bytes), 'S', 'a', (old-style string) 'V' (void)
+        #       not handled datetime and timedelta both map to datetime
+        #       (is timedelta handled?)
+        _k = _DtypeKind
+        _np_kinds = {
+            "i": _k.INT,
+            "u": _k.UINT,
+            "f": _k.FLOAT,
+            "b": _k.BOOL,
+            "U": _k.STRING,
+            "M": _k.DATETIME,
+            "m": _k.DATETIME,
+        }
+        kind = _np_kinds.get(dtype.kind, None)
+        if kind is None:
+            # Not a NumPy/CuPy dtype. Check if it's a categorical maybe
+            if isinstance(dtype, cudf.CategoricalDtype):
+                kind = _k.CATEGORICAL
```

Comment:
I'd rather use the module-level name rather than a local alias `_k`.
```suggestion
                kind = _DtypeKind.CATEGORICAL
```